test/git_issue_368.cpp

```diff
@@ -0,0 +1,63 @@
+///////////////////////////////////////////////////////////////////////////////
+//  Copyright 2021 Christopher Kormanyos. Distributed under the Boost
+//  Software License, Version 1.0. (See accompanying file
+//  LICENSE_1_0.txt or copy at http://www.boost.org/LICENSE_1_0.txt)
+
+#include <boost/multiprecision/cpp_dec_float.hpp>
+
+#include "test.hpp"
+
+namespace local
+{
+  template<typename DecimalType>
+  void test_ceil_rounding()
+  {
+    using decimal_type = DecimalType;
+
+    bool result_is_ok = true;
+
+    for(std::uint32_t lo_index = 101U; lo_index < 1010U; lo_index += 7U)
+    {
+      for(std::uint32_t hi_index = 10001U; hi_index < 100010U; hi_index += 17U)
+      {
+        const std::uint32_t lo_hi = lo_index * hi_index;
+
+        const decimal_type a = decimal_type { lo_hi } / decimal_type { lo_index };
+        const decimal_type b = ceil(a);
+
+        const std::string str_a = boost::lexical_cast<std::string>(a);
+        const std::string str_b = boost::lexical_cast<std::string>(b);
+
+        result_is_ok &= (str_a == str_b);
```

Comment:
Questions: why are we comparing strings rather than values, and should we also test inputs which are not integer?  Negative values as well, plus floor too I suspect ;)

Reply:
> why are we comparing strings rather than values, 

In this PR, there is rounding on `floor`, `ceil` and cast-to-int only. I removed the use of strings. But the comparison with, say, the result of `floor` or `ceil` is only true if the value being compared is casted to int prior to comparison. This is now done without strings in the candidate code of the PR's test file.

> we also test inputs which are not integer

It seems like there are already-existing rounding tests which do this.

> Negative values as well, plus floor too I suspect

Understood. Done in resent push(es).